src/main/java/ch/njol/skript/ScriptLoader.java

```diff
@@ -158,6 +159,50 @@ public static boolean getCallPreloadEvent() {
 	public static Set<SkriptAddon> getPreloadListeners() {
 		return Collections.unmodifiableSet(preloadListeners);
 	}
+	
+	/**
+	 * If true, a {@link PostStructureCreateEvent} will be called
+	 * right after the structure of a script is generated,
+	 * but before the script is parsed.
+	 */
+	private static boolean callPostStructureEvent;
+
+	/**
+	 * A set of all the SkriptAddons that have called
+	 * {@link ScriptLoader#setCallPostStructureEvent(boolean, SkriptAddon)}
+	 * with true.
+	 */
+	private static Set<SkriptAddon> postStructureListeners = new HashSet<>();
+
+	/**
+	 * Sets {@link ScriptLoader#callPostStructureEvent} to the provided boolean,
+	 * and adds/removes the provided SkriptAddon from {@link ScriptLoader#postStructureListeners}
+	 * depending on the provided boolean (true adds, false removes).
+	 * @param state The new value for {@link ScriptLoader#postStructureListeners}
+	 * @param addon A non-null SkriptAddon
+	 */
+	public static void setCallPostStructureEvent(boolean state, SkriptAddon addon) {
+		Validate.notNull(addon);
+		callPostStructureEvent = state;
+		if (state)
+			postStructureListeners.add(addon);
+		else
+			postStructureListeners.remove(addon);
+	}
+
+	public static boolean getCallPostStructureEvent() {
+		return callPostStructureEvent;
+	}
+
+	/**
+	 * Returns an unmodifiable list of all the addons
+	 * that have called {@link ScriptLoader#setCallPostStructureEvent(boolean, SkriptAddon)}
+	 * with true.
+	 */
+	@SuppressWarnings("null")
+	public static Set<SkriptAddon> getPostStructureListeners() {
+		return Collections.unmodifiableSet(postStructureListeners);
+	}

```

Comment:
I don't understand the use case for this here, why not always call the event. If nobody is listening to it, there's very little overhead anyway (especially on Paper). The Set of addons seems kind of pointless since they're really not useful.

Reply:
I agree with this. I was looking at this last night (we talked about it in discord a little), and this seemed kind of pointless. It also seems like an addon could disable the event being called, which could break other addons that depend on it.

Reply:
it's to be consistent with how i implemented the preload event and i implemented the preload event that way because there is a similar toggle to not call function events. looking back though i wouldn't have done this now